geopandas/plotting.py

```diff
@@ -91,17 +91,26 @@ def plot_polygon_collection(
         )
     from matplotlib.collections import PatchCollection
 
-    geoms, values = _flatten_multi_geoms(geoms, values)
-    if None in values:
-        values = None
+    geoms, multiindex = _flatten_multi_geoms(geoms, range(len(geoms)))
+    if values is not None:
+        values = [values[i] for i in multiindex]
```

Comment:
maybe we can use `np.take` here? (`np.take(values, multiindex)`) That should work for list-input as well, and will be faster for bigger arrays

Reply:
and same for occurrence below

Reply:
Sure, didn't know of `np.take`.